src/psyclone/f2pygen.py

```diff
@@ -911,6 +913,17 @@ def _check_initial_values(dtype, values):
                     raise RuntimeError(
                         "Initial value of '{0}' for a real variable is "
                         "invalid or unsupported".format(val))
+        elif dtype == "character":
```

Comment:
The checks in _check_initial values should be encapsulated in each of the relevant sub-classes rather than the base class. Could you please move these. The method here should then be abstract.

Reply:
Oh yes, well spotted. Tidied it up now.